visualization/app/codeCharta/ui/copyToClipboardButton/copyToClipboardButton.component.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { TestBed } from "@angular/core/testing"
+import { CopyToClipboardButtonComponent } from "./copyToClipboardButton.component"
+import { CopyToClipboardButtonModule } from "./copyToClipboardButton.module"
+
+describe("CopyToClipboardButtonComponent", () => {
+	let component: CopyToClipboardButtonComponent
+
+	beforeEach(() => {
+		const test = TestBed.configureTestingModule({
+			imports: [CopyToClipboardButtonModule]
+		})
+		component = test.createComponent<CopyToClipboardButtonComponent>(CopyToClipboardButtonComponent).componentInstance
+		component["service"].getClipboardText = () => "Magic Monday"
+
+		//clipboard does not exist in jest's JSdom, see https://stackoverflow.com/questions/62351935/how-to-mock-navigator-clipboard-writetext-in-jest
+		Object.assign(navigator, { clipboard: { writeText: jest.fn() } })
+	})
+
+	describe("copyToClipboardButtonComponent", () => {
+		it("should write text to clipboard by using text from service", async () => {
+			const clipboard = navigator.clipboard
+			const writeToClipboardSpy = jest.spyOn(clipboard, "writeText")
+			const getClipboardTextSpy = jest.spyOn(component["service"], "getClipboardText")
+
+			await component.copyNamesToClipBoard()
+
+			expect(writeToClipboardSpy).toBeCalledWith("Magic Monday")
+			expect(getClipboardTextSpy).toBeCalled()
```

Comment:
Do we care if this was called, as long as the clipboard was written with the correct text?
```suggestion
			expect(writeToClipboardSpy).toBeCalledWith("Magic Monday")
```

Reply:
In my opinion: yes. Because the method hast two jobs: 1. Get the clipboard text by calling the service method, 2. Write the result to clipboard.